### ISSUE TITLE
http://localhost:3003/test-page no image

### ISSUE THREAD

**Issue by madshobye** [jimmynicol/image-resizer#33]
I tried to run your code locally but the test page does not show and image. E.g. I have no idea where:

http://localhost:3003/elocal/test/sample_images/image1.jpg

Is located. I am simply a little unsure how to configure the image repository? It is not clear to me how one uses this - e.g. where does the images come from. My hope was that I could supply it with a http url and it would do the resizing for me but it seems like i need som sort of repository?

**Comment by jimmynicol** [comment]
Hi @madshobye,

Did you run the code as the repo, or did you create a new working instance with the command line tools (`image-resizer new`)?

Ostensibly you should be using the latter option. It will do a number of things for you, one of which is setting a local path to the sample files so you can get the test page working. The environment config var you are looking for to change this is `LOCAL_FILE_PATH` (https://github.com/jimmynicol/image-resizer/blob/master/src/config/environment_vars.js#L53).

Out of the box `image-resizer` can work either off of a local directory or its primary use case is an S3 bucket. I decided it wasnt a way I wanted to head in order to pull down any image from the internet to process it. You can do this however in a custom source file if you want to try this yourself. Here is a thread about this: https://github.com/jimmynicol/image-resizer/issues/16.

Hope this helps, let me know if you are still having problems.

**Comment by madshobye** [comment]
Thank you for your quick reply. I will look into creating my own source, but I cannot get the basic setup to work.

I followed your guide (the latter option). The file is there but it cannot read the read the file?

The path is:

LOCAL_FILE_PATH=/Users/[..]/nodejsimageresizer/node_modules/image-resizer

This is the call: 

http://localhost:3003/s200-elocal/test/sample_images/image1.jpg

This the error:

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: write EPIPE
    at errnoException (net.js:905:11)
    at Object.afterWrite (net.js:721:19)
[gulp] [nodemon] app crashed - waiting for file changes before starting...

If I call something invalid I like:

http://localhost:3003/s200-elocal/test/sample_images/imakkge1.jpg

I get the following:
[resizer] new stream created! 
[resizer] not valid image format 

Which is surprising since I would expect it to check for its existence before tjecking the format?

**Comment by madshobye** [comment]
Ok figured it out now. Had to install imagemagick or graphicmagic.

**Comment by madshobye** [comment]
BTW. This should work for downloading via http:

'use strict';

var stream, util, request, env;

stream  = require('stream');
util    = require('util');
request = require('request');
env    = require('../../node_modules/image-resizer/src/config/environment_vars');



function contentLength(bufs){
  return bufs.reduce(function(sum, buf){
    return sum + buf.length;
  }, 0);
}

function httpgrab(image){
  /* jshint validthis:true */
  if (!(this instanceof httpgrab)){
    return new httpgrab(image);
  }
  stream.Readable.call(this, { objectMode : true });
  this.image = image;
  this.ended = false;

  // set the expiry value to the shorter value
  this.image.expiry = env.IMAGE_EXPIRY_SHORT;
}

util.inherits(httpgrab, stream.Readable);

httpgrab.prototype._read = function(){
  var _this = this,
      url, videoId,
      imgStream,
      bufs = [];

  if ( this.ended ){ return; }

  // pass through if there is an error on the image object
  if (this.image.isError()){
    this.ended = true;
    this.push(this.image);
    return this.push(null);
  }


  console.log('hsdf');
  videoId = this.image.image.split('.')[0];
  url = 'http://' + this.image.path.replace(/^\//,'');
  console.log(url);
  this.image.log.time('httpgrab');

  imgStream = request.get(url);
  imgStream.on('data', function(d){ bufs.push(d); });
  imgStream.on('error', function(err){
    _this.image.error = new Error(err);
  });
  imgStream.on('end', function(){
    _this.image.log.timeEnd('httpgrab');
    _this.image.contents = Buffer.concat(bufs);
    _this.image.originalContentLength = contentLength(bufs);
    _this.ended = true;
    _this.push(_this.image);
    _this.push(null);
  });

};


module.exports = httpgrab;

**Comment by jimmynicol** [comment]
@madshobye I just saw that pastebin has been removed but I had it copied locally. See this gist: https://gist.github.com/jimmynicol/eeb3c7fc3872ea24dc3d

I had used it just for quick testing so it does work. I would be very careful though as it does open your app up to abuse as anyone can use your service to process images. The original intent of the app to process S3 images protects you some what from this as people can only process images you control.
